lib/garage/tracer.rb

```diff
@@ -0,0 +1,90 @@
+module Garage
+  module Tracer
+    extend self
+    extend Forwardable
+
+    def_delegators :tracer, :start, :inject_trace_contect, :record_http_request, :record_http_response
+
+    private
+
+    def tracer
+      case Garage.configuration.tracing[:tracer]
```

Comment:
Aha, it was actually "tracers are plugable and users who want to use custom tracer can sends PR to garage".

+1 for your idea, setting `Tracer` object directly.